galaxy_importer/config.py

```diff
@@ -0,0 +1,86 @@
+# (c) 2012-2020, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import configparser
+import os
+
+
+FILENAME = 'galaxy-importer.cfg'
+FILE_LOCATIONS = [
+    f'/etc/galaxy-importer/{FILENAME}',
+    os.path.join(os.path.dirname(__file__), FILENAME),
+]
+IMPORTER_INI_SECTION = 'galaxy-importer'
+
+
+class Config(object):
+    """Configuration for galaxy-importer."""
+
+    DEFAULTS = {
+        'log_level_main': 'INFO',
+        'run_ansible_test': False,
+        'infra_pulp': False,
+        'infra_osd': False,
+    }
+    IS_LOADED = False
+    CONFIG_DATA = {}
+
+    def __init__(self):
+        """Load config once and store inside class variable CONFIG_DATA.
+        Populate instance from class variable CONFIG_DATA."""
+
+        if not Config.IS_LOADED:
+            Config.IS_LOADED = True
```

Comment:
Often changing class variables (IS_LOADED, CONFIG_DATA) at runtime can lead to odd and difficult to debug behavior. I'm not sure I would be able to predict the expected behavior is this class is instantiated multiple times
especially after Config.CONFIG_DATA is updated with config_file_data.  It is singleton-ish but for lack of better words, it seems kind of odd and troubleprone to me.

And again, if this is something that is required to work around some ansible-test weirdness, it may be unavoidable.